app/models/eligible_ey_provider.rb

```diff
@@ -25,10 +25,14 @@ def self.csv
   end
 
   def self.eligible_email?(email_address)
-    if email_address.present?
-      where(primary_key_contact_email_address: email_address).or(
-        where(secondary_contact_email_address: email_address)
-      ).exists?
-    end
+    email_address.present? && for_email(email_address).exists?
```

Comment:
```suggestion
    for_email(email_address).exists?
```
does the `email_address.present?` check in `for_email` mean that we don't need to the same check here?